### PR TITLE
[TechDocs] Fix footer links width for newer versions of mkdocs-material

### PR DESCRIPTION
**Pull request by gusevda**
## Hey, I just made a Pull Request!

Fixes https://github.com/backstage/backstage/issues/19383.

There is a [width restriction](https://github.com/backstage/backstage/blob/master/plugins/techdocs/src/reader/transformers/styles/rules/layout.ts#L123-L125) for next/previous links in a footer. But in newer version of `mkdocs-material` those links have a different class name -`md-footer__link` instead of `md-footer-nav__link` (details are [here](https://squidfunk.github.io/mkdocs-material/upgrade/#partialsfooterhtml_1)).

In this PR I added a new class name to the existing styles transformer to fix the issue.

Before:
Screenshot 2023-08-15 at 19 11 08

After:
Screenshot 2023-08-15 at 19 57 27

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by agentbellnorm — APPROVED** [review]

**Comment by agentbellnorm** [comment]
Looks great, thanks! I guess we just need @backstage/maintainers blessing and we should be good to go!

**Review by benjdlambert — APPROVED** [review]

**Comment by benjdlambert** [comment]
Thanks! :tada:

**Pull request merged**
